src/samplers/pmmh.jl

```diff
@@ -163,63 +160,3 @@ sample(model::Function, alg::PMMH;
 
     c
 end
-
-function rand_truncated(dist, lowerbound, upperbound)
-    notvalid = true
-    x = 0.0
-    while (notvalid)
-        x = rand(dist)
-        notvalid = ((x < lowerbound) | (x > upperbound))
-    end
-    return x
-end
-
-assume(spl::Sampler{PMMH}, dist::Distribution, vn::VarName, vi::VarInfo) = begin
-    if isempty(spl.alg.space) || vn.sym in spl.alg.space
-      vi.index += 1
-      if ~haskey(vi, vn) #NOTE: When would that happens ??
-        r = rand(dist)
-        push!(vi, vn, r, dist, spl.alg.gid)
-        spl.info[:cache_updated] = CACHERESET # sanity flag mask for getidcs and getranges
-      elseif vn.sym in keys(spl.alg.proposals) # Custom proposal for this parameter
-        oldval = getval(vi, vn)[1]
-        proposal = spl.alg.proposals[vn.sym](oldval)
-        if typeof(proposal) == Distributions.Normal{Float64} # If Gaussian proposal
-          σ = std(proposal)
-          lb = support(dist).lb
-          ub = support(dist).ub
-          stdG = Normal()
-          r = rand_truncated(proposal, lb, ub)
-          # cf http://fsaad.scripts.mit.edu/randomseed/metropolis-hastings-sampling-with-gaussian-drift-proposal-on-bounded-support/
-          spl.info[:proposal_prob] += log(cdf(stdG, (ub-oldval)/σ) - cdf(stdG,(lb-oldval)/σ))
-          spl.info[:proposal_prob] -= log(cdf(stdG, (ub-r)/σ) - cdf(stdG,(lb-r)/σ))
-      else # Other than Gaussian proposal
-          r = rand(proposal)
-          if (r < support(dist).lb) | (r > support(dist).ub) # check if value lies in support
-            spl.info[:violating_support] = true
-            r = oldval
-          end
-          spl.info[:proposal_prob] -= logpdf(proposal, r) # accumulate pdf of proposal
-          reverse_proposal = spl.alg.proposals[vn.sym](r)
-          spl.info[:proposal_prob] += logpdf(reverse_proposal, oldval)
-        end
-        spl.info[:new_prior_prob] += logpdf(dist, r) # accumulate pdf of prior
-      else # Prior as proposal
-        r = rand(dist)
-      end
-      setval!(vi, vectorize(dist, r), vn)
-      setgid!(vi, spl.alg.gid, vn)
-      r
-    else
-      vi[vn]
-    end
-end
-
-assume{D<:Distribution}(spl::Sampler{PMMH}, dists::Vector{D}, vn::VarName, var::Any, vi::VarInfo) =
-  error("[Turing] PMMH doesn't support vectorizing assume statement")
-
-observe(spl::Sampler{PMMH}, d::Distribution, value::Any, vi::VarInfo) =
```

Comment:
Just for my understanding, what happens if you call `observe(spl::Sampler{PMMH}, ...)` now? We don't have `PMMH <: MH` so how would it know which method to call?

Reply:
`observe(spl::Sampler{PMMH}, ...)` would be called inside`model(vi=vi, sampler=spl::Sampler{PMMH})`, which is never done. `model` is only ran with `MH` and `SMC` not `PMMH` itself.

Reply:
ok cheers